scanner/src/main/kotlin/scanners/scancode/ScanCodeResultParser.kt

```diff
@@ -252,47 +253,58 @@ internal fun replaceLicenseKeys(licenseExpression: String, replacements: Collect
 private const val LICENSES_WITH_EXCEPTIONS_TOLERANCE_LINES = 5
 
 internal fun associateLicensesWithExceptions(findings: List<LicenseFinding>): List<LicenseFinding> {
-    val otherFindings = mutableListOf<LicenseFinding>()
+    val (exceptions, licenses) = findings.partition { SpdxLicenseException.forId(it.license.toString()) != null }
 
-    val apacheLicenses = mutableListOf<LicenseFinding>()
-    val llvmExceptions = mutableListOf<LicenseFinding>()
+    val remainingExceptions = exceptions.toMutableList()
+    val fixedLicenses = licenses.toMutableList()
 
-    findings.forEach {
-        when (it.license.toString()) {
-            "Apache-2.0" -> apacheLicenses += it
-            "LLVM-exception" -> llvmExceptions += it
-            else -> otherFindings += it
+    val e = remainingExceptions.iterator()
```

Comment:
Using `e` here makes the code look a bit like it would handle actual throwable exceptions, not license exceptions.

Reply:
Renamed to `i` instead 😁 (Unless you come up with a better name, but I'd like to keep it short.)

Reply:
...although IMO `e.remove()` read nicer than `i.remove()`.